plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/UnnecessaryOptInAnnotationInspection.kt

```diff
@@ -113,7 +114,7 @@ class UnnecessaryOptInAnnotationInspection : AbstractKotlinInspection() {
             annotationEntry.getOwner()?.accept(OptInMarkerVisitor(), markerProcessor)
 
             val unusedMarkers = resolvedMarkers.filter { markerProcessor.isUnused(it.fqName) }
-            if (annotationEntry.valueArguments.size == unusedMarkers.size) {
+            if (annotationEntryArguments.isNotEmpty() && annotationEntryArguments.size == unusedMarkers.size) {
```

Comment:
We shouldn't check an annotation entry without arguments at all. This fast check should be done at the begining of `annotationEntryVisitor`

Reply:
> We shouldn't check an annotation entry without arguments at all. This fast check should be done at the begining of `annotationEntryVisitor`

Done